AnomalyDetection/LibLinear/src/main/java/org/tribuo/anomaly/liblinear/LibLinearAnomalyTrainer.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.tribuo.anomaly.liblinear;
+
+import com.oracle.labs.mlrg.olcut.config.Config;
+import com.oracle.labs.mlrg.olcut.util.Pair;
+import org.tribuo.Dataset;
+import org.tribuo.Example;
+import org.tribuo.ImmutableFeatureMap;
+import org.tribuo.ImmutableOutputInfo;
+import org.tribuo.Trainer;
+import org.tribuo.anomaly.Event;
+import org.tribuo.anomaly.liblinear.LinearAnomalyType.LinearType;
+import org.tribuo.common.liblinear.LibLinearModel;
+import org.tribuo.common.liblinear.LibLinearTrainer;
+import org.tribuo.provenance.ModelProvenance;
+import de.bwaldvogel.liblinear.FeatureNode;
+import de.bwaldvogel.liblinear.Linear;
+import de.bwaldvogel.liblinear.Model;
+import de.bwaldvogel.liblinear.Parameter;
+import de.bwaldvogel.liblinear.Problem;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Logger;
+
+/**
+ * A {@link Trainer} which wraps a liblinear-java anomaly detection trainer using a one-class SVM.
+ *
+ * See:
+ * <pre>
+ * Fan RE, Chang KW, Hsieh CJ, Wang XR, Lin CJ.
+ * "LIBLINEAR: A library for Large Linear Anomaly"
+ * Journal of Machine Learning Research, 2008.
+ * </pre>
+ * and for the original algorithm:
+ * <pre>
+ * Cortes C, Vapnik V.
+ * "Support-Vector Networks"
+ * Machine Learning, 1995.
+ * </pre>
+ */
+public class LibLinearAnomalyTrainer extends LibLinearTrainer<Event> {
+
+    private static final Logger logger = Logger.getLogger(LibLinearAnomalyTrainer.class.getName());
+
+    @Config(description = "Nu parameter in one class SVM.")
+    private double nu = 0.5;
+
+    /**
+     * Creates a trainer using the default values (ONECLASS_SVM, 1, 0.1, 0.5).
```

Comment:
The JavaDoc doesn't match the implementation here (for maxIterations) or in the next few constructors

Reply:
Ooops, yeah I should put max iterations in.